specification/security/resource-manager/Microsoft.Security/preview/2020-08-06-preview/insights.json

```diff
@@ -0,0 +1,254 @@
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/InsightsList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{resourceId}/providers/Microsoft.Security/insights/{insightName}": {
+      "get": {
+        "x-ms-examples": {
+          "Get classification insights of an Azure storage resource": {
+            "$ref": "./examples/Insights/GetStorageClassificationInsights_example.json"
+          }
+        },
+        "tags": [
+          "Insights"
+        ],
+        "description": "Insights of a specific resource",
+        "operationId": "Insights_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/InsightName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/InsightsModel"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "InsightsList": {
+      "type": "object",
+      "readOnly": true,
+      "description": "Insights list response.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of insights.",
+          "items": {
+            "$ref": "#/definitions/InsightsModel"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URI to fetch the next page."
+        }
+      }
+    },
+    "InsightsModel": {
+      "type": "object",
+      "description": "Insights of a specific resource.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Insights details data",
+          "$ref": "#/definitions/InsightsProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "InsightsProperties": {
+      "type": "object",
+      "description": "Describes properties of resource insights.",
+      "properties": {
+        "associatedResource": {
```

Comment:
isn't this the same as the resource id in the url? if so, why is it needed?

Reply:
It's the same. The resourceId is part of the URI.
We might want to support a List operation later, so we considered adding it here.
It still can be extracted from the resourceId, but I don't see any harm of keeping it here

Reply:
@ranisha2 It's not the common pattern followed in ARM. I would also recommend to remove this redundant property. As this property is not readOnly and configurable by user, you will need to do extra checks like 'resourceId' in the URL should be same as that of the value provided in resource payload.

Reply:
@RamyasreeChakka Actually it's a read only property. I will set it as a read only.